Cesium3DTiles/src/RasterizedPolygonsOverlay.cpp

```diff
@@ -0,0 +1,222 @@
+
+#include "Cesium3DTiles/RasterizedPolygonsOverlay.h"
+#include "Cesium3DTiles/BoundingVolume.h"
+#include "Cesium3DTiles/RasterOverlayTileProvider.h"
+#include "Cesium3DTiles/spdlog-cesium.h"
+#include "CesiumAsync/AsyncSystem.h"
+#include "CesiumAsync/IAssetAccessor.h"
+#include "CesiumGeospatial/GlobeRectangle.h"
+#include "CesiumUtility/IntrusivePointer.h"
+#include <memory>
+#include <string>
+
+namespace Cesium3DTiles {
+namespace {
+void rasterizePolygons(
+    CesiumGltf::ImageCesium& image,
+    const CesiumGeospatial::GlobeRectangle& rectangle,
+    const std::string& textureTargetName,
+    const std::vector<CartographicSelection>& cartographicSelections) {
```

Comment:
Since cartographicSelections is just a 2D polygon, I think maybe rasterize whole polygon is more efficient than triangles, but I'm not sure on that. But in case you need to try, the algorithm for 2D polygon rasterization is here http://alienryderflex.com/polygon_fill/